src/commit-body.ts

```diff
@@ -1,21 +1,83 @@
+import Ajv, {JTDParser, JTDSchemaType} from 'ajv/dist/jtd'
+import {InvalidCommitBodyError} from './errors'
 import {Message} from './message'
+import isEqual from 'lodash.isequal'
+
+interface CommitBodyMetaData {
+  job_number?: number
+  job_commit?: string
+}
+
+interface CommitBodyData {
+  namespace: 'git-queue.commit-body'
+  version: number
+  metadata: CommitBodyMetaData
+  payload: string
+}
+
+const CommitBodySchema: JTDSchemaType<CommitBodyData> = {
+  properties: {
+    namespace: {enum: ['git-queue.commit-body']},
+    version: {type: 'int32'},
+    payload: {type: 'string'},
+    metadata: {
+      optionalProperties: {
+        job_number: {type: 'int32'},
+        job_commit: {type: 'string'}
+      },
+      additionalProperties: true
+    }
+  },
+  additionalProperties: true
+}
 
 export class CommitBody {
-  text: string
+  body: CommitBodyData
+  private static parse: JTDParser<CommitBodyData> = new Ajv().compileParser(
+    CommitBodySchema
+  )
 
   constructor(text: string) {
-    this.text = text
+    this.body = this.getParsedBodyData(text)
   }
 
   static fromMessage(message: Message): CommitBody {
-    return new CommitBody(message.getPayload())
+    return new CommitBody(
+      JSON.stringify({
+        namespace: 'git-queue.commit-body',
+        version: 1,
+        metadata: {
+          job_commit: message.getJobRef().getHash()
+        },
+        payload: message.getPayload()
+      })
+    )
+  }
+
+  getParsedBodyData(text): CommitBodyData {
+    const parsedBody = CommitBody.parse(text)
+    if (this.isCommitBodyData(parsedBody)) {
+      return parsedBody
+    } else {
+      throw new InvalidCommitBodyError(text)
+    }
+  }
+
+  isCommitBodyData(
+    object: CommitBodyData | undefined
+  ): object is CommitBodyData {
+    return typeof object != 'undefined'
   }
 
   toString(): string {
-    return this.text
+    return JSON.stringify(this.body)
+  }
+
+  getPayload(): string {
+    return this.body?.payload
   }
 
   equalsTo(other: CommitBody): boolean {
-    return this.text === other.text
+    return isEqual(this, other)
```

Comment:
@yeraydavidrodriguez I think we agree on having our own comparison function, right?. Instead of using `lodash.isequal`.

Reply:
Added custom comparison function